scripts/bootloader/validation_data.py

```diff
@@ -12,43 +12,29 @@
 import struct
 import ecdsa
 
-VERBOSE = False
-
-
-def verbose_print(*printstrs):
-    if VERBOSE:
-        print(*printstrs)
-
 
 def get_hash(input_hex):
     firmware_bytes = input_hex.tobinstr()
-    verbose_print("firmware len: %d (0x%x)" % (len(firmware_bytes), len(firmware_bytes)))
     return hashlib.sha256(firmware_bytes).digest()
 
 
 def get_validation_data(signature_bytes, input_hex, public_key, magic_value, pk_hash_len):
     hash_bytes = get_hash(input_hex)
     public_key_bytes = public_key.to_string()
-    public_key_hash_bytes = hashlib.sha256(public_key_bytes).digest()[:pk_hash_len]
 
-    verbose_print("magic value (hex) (len: %d): %s" % (len(magic_value), bytes.hex(magic_value)))
-    verbose_print("firmware hash (hex) (len: %d): %s" % (len(hash_bytes), bytes.hex(hash_bytes)))
-    verbose_print("public key (hex) (len: %d): %s" % (len(public_key_bytes), bytes.hex(public_key_bytes)))
-    verbose_print("truncated public key hash (hex) (len: %d): %s" % (len(public_key_hash_bytes), bytes.hex(public_key_hash_bytes)))
-    verbose_print("signature (hex) (len: %d): %s" % (len(signature_bytes), bytes.hex(signature_bytes)))
-
-    verbose_print("signature ok? ", public_key.verify(signature_bytes, hash_bytes, hashfunc=hashlib.sha256))
+    # Will raise an exception if it fails
+    public_key.verify(signature_bytes, hash_bytes, hashfunc=hashlib.sha256)
 
     validation_bytes = magic_value
     validation_bytes += struct.pack('I', input_hex.addresses()[0])
     validation_bytes += hash_bytes
-    validation_bytes += public_key_bytes
+    validation_bytes += public_key_bytes[:pk_hash_len]
```

Comment:
The truncation is to be applied to the hash of the PK, not the PK itself.
Come to think of it, the pk_hash_len is probably not needed at all in this script

Reply:
If the whole public key is in there, why do we have the hash there?

Reply:
Good catch, updated, and updated commit message, please verify that I didn't misunderstand.

Reply:
This is really just a cleanup PR now.

Reply:
The commit message doesn't really match the diff IMO.

Reply:
updated